odd-platform-api/src/main/java/com/provectus/oddplatform/service/TagServiceImpl.java

```diff
@@ -6,14 +6,40 @@
 import com.provectus.oddplatform.mapper.TagMapper;
 import com.provectus.oddplatform.model.tables.pojos.TagPojo;
 import com.provectus.oddplatform.repository.TagRepository;
+import java.util.List;
+import java.util.stream.Collectors;
 import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
 
 @Service
 public class TagServiceImpl
     extends AbstractCRUDService<Tag, TagsResponse, TagFormData, TagFormData, TagPojo, TagMapper, TagRepository>
     implements TagService {
 
-    public TagServiceImpl(final TagMapper entityMapper, final TagRepository entityRepository) {
-        super(entityMapper, entityRepository);
+    private final TagRepository repository;
+
+    public TagServiceImpl(final TagMapper entityMapper, final TagRepository repository) {
+        super(entityMapper, repository);
+        this.repository = repository;
+    }
+
+    public Mono<TagsResponse> listMostPopular(final String query, final int page, final int size) {
+        return Mono
+            .fromCallable(() -> repository.listMostPopular(query, page, size))
+            .map(tags -> {
+                final List<Tag> items = tags.stream()
```

Comment:
Whole mapping logic can be moved into the mapper

Reply:
Done